src/core/path.c

```diff
@@ -494,6 +497,12 @@ static void path_enter_running(Path *p) {
         if (unit_stop_pending(UNIT(p)))
                 return;
 
+        if (!ratelimit_below(&p->trigger_limit)) {
+                log_unit_warning(UNIT(p), "Trigger limit hit, refusing further activation.");
+                path_enter_dead(p, PATH_FAILURE_TRIGGER_LIMIT_HIT);
```

Comment:
will it become runnable again automatically later with this?

Reply:
It won't unless the path unit is started again, but if we hit this at all, it very likely means that the path unit is triggering a busy loop in systemd so we don't actually want it to trigger again until that's fixed.

Reply:
This is an issue though - for example think about a service using the new psi based conditions. Things shouldn't fail forever if a condition is missed. Can't we add a timer event to automatically clean the trigger limit after a while?

Reply:
Is it an issue though if you consider that currently, this situation would cause PID1 to busy loop when a condition is missed? The trigger limits for socket units don't have such remediation either unless I missed something.

I figure at the moment, using conditions with path units containing PathExists is completely broken since as soon as conditions fail, you end up busy looping PID 1. There's two problems here, one is that it's possible to cause a busy loop in PID 1 by combined usage of conditions and PathExists. The second problem is figuring out how we should define the interaction between PathExists and failed conditions.

This PR solves the first problem but not the second one. I still think it's an improvement over what we have since we solve the first problem. The second problem is IMO best solved in a second PR and can be delayed to v251.

Reply:
I agree that this is better than the current situation, however we really need to fix the side effect too, as it makes the combination of event triggered services and conditions way less powerful than it could be